File: app/src/main/java/com/dm/bomber/bomber/AuthProxy.java
package com.dm.bomber.bomber;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.net.InetSocketAddress;
import java.net.Proxy;

import okhttp3.Authenticator;
import okhttp3.Request;
import okhttp3.Response;
import okhttp3.Route;

public class AuthProxy extends Proxy implements Authenticator {
    private final String credential;

    public AuthProxy(Proxy.Type type, InetSocketAddress sa, String credential) {
        super(type, sa);

        this.credential = credential;
    }

    @Nullable
    @Override
    public Request authenticate(@Nullable Route route, @NonNull Response response) {
        if (credential != null)
            return response.request().newBuilder()
                    .header("Proxy-Authorization", credential)
                    .build();

        return null;
    }
}

File: app/src/main/java/com/dm/bomber/worker/AuthProxy.kt
package com.dm.bomber.worker

import okhttp3.Authenticator
import okhttp3.Request
import okhttp3.Response
import okhttp3.Route
import java.net.InetSocketAddress
import java.net.Proxy

class AuthProxy(
    type: Type?,
    sa: InetSocketAddress?,
    private val credential: String?
) : Proxy(type, sa), Authenticator {
    override fun authenticate(route: Route?, response: Response): Request? {
        return if (credential != null) response.request.newBuilder()
            .header("Proxy-Authorization", credential)
            .build() else null
    }
}
